src/main/java/org/interledger/cryptoconditions/PreimageSha256Fulfillment.java

```diff
@@ -12,30 +13,22 @@
  * @author adrianhopebailie
  *
  */
-public class PreimageSha256Fulfillment implements Fulfillment {
+public class PreimageSha256Fulfillment extends FulfillmentBase {
+
+	public static final ConditionType CONDITION_TYPE = ConditionType.PREIMAGE_SHA256;
+
+	public PreimageSha256Fulfillment(ConditionType type, byte[] payload) {
+		super(type, payload);
+	}
 
-	private static ConditionType CONDITION_TYPE = ConditionType.PREIMAGE_SHA256;
-	
 	private static EnumSet<FeatureSuite> BASE_FEATURES = EnumSet.of(
 			FeatureSuite.SHA_256, 
 			FeatureSuite.PREIMAGE
 		);
 
-	private byte[] preimage;
```

Comment:
I was thinking about Fulfillments as something that can be "printed on paper", so completely immutable once created. The idea was to force immutability as-soon-as-possible (in the constructor), but I'm noticing that different fulfillments are so different in nature (according to the "inputs" needed and internal algorithms to create and validate the condition) that using a FulfillmentBase will just work painfully. I think creating an immutable object at construction time is still possible with the help of some external collaborator object for each different Fulfillment that avoids the problems with inheritance, but I'm not yet there. This collaborator interface (and implementing classes) would get in charge of generating conditions and validations and each one could store private data that the Fulfillment could then ignore, keeping just the type and payload (all that needed to "print on paper"). This could have some potential securities advantages since the collaborator instances could be removed from memory once the Fulfillment is fulfilled and have a short live-time period (just enough to create the fulfillment data). Those collaborators would store the secret (private key), so the smaller live-time the better. The fulfillment could be stay in memory, cached, logged or serialized with no risk of leaking the secret (This is all theory, only code can confirm). Since we are still in the initial phase I don't have any issues with dropping this PR or continuing with it. Use the code base that is better for your purposes.